clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java

```diff
@@ -273,10 +279,20 @@ public void disconnected(ClientDescriptor clientDescriptor) {
 
   @Override
   public EhcacheEntityResponse invoke(ClientDescriptor clientDescriptor, EhcacheEntityMessage message) {
-    switch (message.getType()) {
-      case LIFECYCLE_OP: return invokeLifeCycleOperation(clientDescriptor, (LifecycleMessage) message);
-      case SERVER_STORE_OP: return invokeServerStoreOperation(clientDescriptor, (ServerStoreOpMessage) message);
-      default: throw new IllegalArgumentException("Unknown message " + message);
+    try {
+      switch (message.getType()) {
+        case LIFECYCLE_OP:
+          return invokeLifeCycleOperation(clientDescriptor, (LifecycleMessage) message);
+        case SERVER_STORE_OP:
+          return invokeServerStoreOperation(clientDescriptor, (ServerStoreOpMessage) message);
+        default:
+          throw new IllegalMessageException("Unknown message : " + message);
+      }
+    } catch (ClusteredEhcacheException e) {
+      return responseFactory.failure(e);
+    } catch (Exception e) {
+      LOGGER.error("Unexpected exception raised during LifeCycle operation: " + e, e);
```

Comment:
You're right; this message should be more generic as it applies to both `LifeCycleOperation` and `ServerStoreOperations`.